src/Orangebeard.Client/AbstractClient.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Orangebeard.Client.Entities;
+using Orangebeard.Client.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+
+namespace Orangebeard.Client
+{
+    abstract public class AbstractClient : IOrangebeardClient
+    {
+        public static readonly string APPLICATION_JSON = "application/json"; //TODO?~ Get this from the MIME Type map.
+
+        protected HttpClient httpClient = new HttpClient();
+
+        public abstract void FinishTestItem(Guid itemId, FinishTestItem finishTestItem);
+        public abstract void FinishTestRun(Guid testRunUUID, FinishTestRun finishTestRun);
+        public abstract void Log(Log log);
+        public abstract void Log(ISet<Log> logs);
+        public abstract void SendAttachment(Attachment attachment);
+        public abstract Guid? StartTestItem(Guid? suiteId, StartTestItem testItem);
+        public abstract Guid? StartTestRun(StartTestRun testRun);
+        public abstract void UpdateTestRun(Guid testRunUUID, UpdateTestRun updateTestRun);
+
+
+        public void InitializeHttpClient(Uri baseUri, string token, string userAgentPostFix)
+        {
+            var httpClientHandler = new HttpClientHandler();
+
+#if !NET45
+                httpClientHandler.ServerCertificateCustomValidationCallback = (message, cert, chain, errors) => { return true; };
+#endif
+
+            httpClient = new HttpClient(httpClientHandler);
+
+            httpClient.BaseAddress = baseUri.Normalize();
+
+            httpClient.DefaultRequestHeaders.Clear();
+            httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + token);
+            //TODO?- This probably isn't needed anymore.
```

Comment:
We should not have todo's / commented out code blocks in the released version

Reply:
Agreed. 
In this particular case I've decided to uncomment the code block in question. It specifies a User-Agent header, and having a User-Agent header is considered a good practice (https://stackoverflow.com/a/24274846/812149, referencing RFC 7231 section 5.5.3).